includes/filters.php

```diff
@@ -117,13 +129,36 @@ function give_akismet( $spam ) {
 	$ignore = array( 'HTTP_COOKIE', 'HTTP_COOKIE2', 'PHP_AUTH_PW' );
 
 	foreach ( $_SERVER as $key => $value ) {
-		if ( ! in_array( $key, (array) $ignore ) ) {
-			$args["$key"] = $value;
+		if ( ! in_array( $key, $ignore, true ) ) {
+			$args[$key] = $value;
 		}
 	}
 
+	$response = give_akismet_spam_check_post( $args );
+	$spam     = 'true' === $response[1];
+
+	// Log spam information.
+	if( $spam ) {
+		give_record_log(
+			sprintf(
+				'Spam flagged for %1$s (%2$s)',
+				$args['comment_author'],
+				$args['comment_author_email']
+			),
+			sprintf(
+				'<p><strong>%1$s</strong><pre>%2$s</pre></p><strong>%3$s</strong><pre>%4$s</pre><p>',
+				__('Request', 'give-zapier'),
+				print_r( $args, true ),
+				__('Response', 'give-zapier'),
+				print_r( $response, true )
+			),
+			0,
+			'spam'
+		);
+	}
+
 	// It will return Akismet spam detect API response.
-	return give_akismet_spam_check( $args );
+	return $spam;
```

Comment:
Maybe a filter here?

Reply:
@DevinWalker adding unnecessary hooks reduces performance.  If you have a case then I can add a filter otherwise no use to provide here. I did not find any scenario where admin wants to change this

Reply:
https://github.com/impress-org/givewp/commit/ac284f535bc3f450e967eec77bbaadc474d2778a